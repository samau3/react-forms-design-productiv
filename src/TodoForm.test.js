import React from "react";
import { render } from "@testing-library/react";

import TEST_DATA from "./_testCommon";
import TodoForm from "./TodoForm";

it("renders without crashing", function () {
    render(<TodoForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

// it("can add a new todo", function () {
//     const { getByLabelText, queryByText } = render(<TodoForm />);

//     // no items yet
//     expect(queryByText("ice cream: 100")).not.toBeInTheDocument();

//     const titleInput = getByLabelText("Title:");
//     const descriptionInput = getByLabelText("Description:");
//     const submitBtn = queryByText("Add a new item!");

//     // fill out the form
//     fireEvent.change(nameInput, { target: { value: "ice cream" } });
//     fireEvent.change(qtyInput, { target: { value: 100 } });
//     fireEvent.click(submitBtn);

//     // item exists!
//     expect(queryByText("ice cream: 100")).toBeInTheDocument();
//   });