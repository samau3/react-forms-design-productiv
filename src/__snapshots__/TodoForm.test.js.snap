// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <div
      class="form-group"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      />
    </div>
    <div
      class="form-group d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</DocumentFragment>
`;
